src/components/AddXpubDialog/AddXpubDialog.tsx

```diff
@@ -0,0 +1,102 @@
+import { CirclePlus } from 'lucide-react';
+import { Button } from '@/components/ui/button.tsx';
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog.tsx';
+import { Label } from '@/components/ui/label.tsx';
+import { Input } from '@/components/ui/input.tsx';
+import React, { useEffect, useState } from 'react';
+import { HD } from '@bsv/sdk';
+import { errorWrapper } from '@/utils';
+import { toast } from 'sonner';
+import { useDebounce } from 'use-debounce';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useSpvWalletClient } from '@/contexts';
+
+export const AddXpubDialog = () => {
+  const [xPriv, setXPriv] = useState<string>('');
+  const [xPub, setXPub] = useState<string>('');
+  const [debouncedXPriv] = useDebounce(xPriv, 500);
+  const queryClient = useQueryClient();
+
+  const { spvWalletClient } = useSpvWalletClient();
+
+  const mutation = useMutation({
+    mutationFn: async (xpub: string) => {
+      return await spvWalletClient!.AdminNewXpub(xpub, {});
+    },
+  });
+
+  const handleXPrivChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setXPriv(event.target.value);
+  };
+
+  useEffect(() => {
+    if (!xPriv) return;
+
+    try {
+      const xPrivHD = HD.fromString(debouncedXPriv);
+      const xPubString = xPrivHD.toPublic().toString();
+      setXPub(xPubString);
+      toast.success('Converted xPriv to xPub');
+    } catch (error) {
+      toast.error('Unable to convert xPriv to xPub');
+      setXPub('');
+    }
+  }, [debouncedXPriv]);
```

Comment:
Now I'm recalling this page. When you paste the xpriv it instantly disappears and the xpub shows...
For me, the UX of this page should be rethinked - because it is weird.

Reply:
In this version xPriv does not disappear. 
<img width="437" alt="Screenshot 2024-06-07 at 10 55 03 AM" src="https://github.com/bitcoin-sv/spv-wallet-admin/assets/150656925/f00a93d6-3f99-4523-87d5-4d96d7658e8a">

Reply:
I was thinking about sth like that. (it's just a conceptional diagram)
![image](https://github.com/bitcoin-sv/spv-wallet-admin/assets/152964795/ede97613-ab9f-47b6-8602-e20306f4f721)

Reply:
It converts automatically and need less interaction from user that has a positive impact on UX